hbase-server/src/main/java/org/apache/hadoop/hbase/regionserver/CompactSplit.java

```diff
@@ -794,13 +795,30 @@ void shutdownLongCompactions(){
   }
 
   public void clearLongCompactionsQueue() {
+    removeFromFilesCompacting(longCompactions);
```

Comment:
If some requests fail to clean up the  compacting files, it may cause some of the compacting files to be removed, but the compacting request has not been remove from the queue

Reply:
yes. Instead of removing all files from file compacting list first and then clearing all compaction requests,  maybe it is better to remove files and compaction request one by one. I will fix it.